features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/btle/btle.cpp

```diff
@@ -298,10 +298,10 @@ static void btle_handler(ble_evt_t *p_ble_evt)
     gattServer.hwCallback(p_ble_evt);
 }
 
-/*! @brief      Callback when an error occurs inside the SoftDevice */
+/*! @brief      Callback when an error occurs inside the SoftDevice or ASSERT in debug*/
 void assert_nrf_callback(uint16_t line_num, const uint8_t *p_file_name)
 {
-    ASSERT_TRUE(false, (void) 0);
+    error("nrf failure at %s:%d", p_file_name, line_num);
```

Comment:
Can you make this change for the NRF51 file too?

Reply:
Oh yeah I forgot about that old NRF51822 platform, do you know why it's still being kept around rather than everyone using the unified nrf5 platform?
 It only takes a couple of simple target definitions to switch.

Reply:
@andrewleech I think the best person to answer that atm would be @marcuschangarm